src/porepy/applications/md_grids/model_geometries.py

```diff
@@ -155,3 +159,97 @@ def meshing_arguments(self) -> dict:
+        # are provided, is to refine by a factor of 2.
+        ratios = self.params.get(
+            "fracture_refinement_ratios", np.ones(len(old_fracture_grids) * 2)
+        )
+
+        # The actual refinement of the fracture grids.
+        new_fracture_grids = [
+            pp.refinement.refine_grid_1d(g=old_grid, ratio=ratio)
+            for old_grid, ratio in zip(old_fracture_grids, ratios)
+        ]
+
+        # Create a mapping between the old and new fracture grids.
+        grid_map = dict(zip(old_fracture_grids, new_fracture_grids))
+
+        # Refine and replace interface grids.
+        # Directly refining the already existing interface grids is not straightforward.
+        # Instead, we create a new mixed-dimensional grid and donate the interface grids
+        # to the original mixed-dimensional grid.
+
+        # First we gather the number of cells in x- and y- direction, as well as the
+        # physical dimensions of the original mixed-dimensional grid:
+        domain = pp.Domain(self.domain.bounding_box)
+        meshing_args = self.meshing_arguments()
+        nx_cells, phys_dims, _ = _preprocess_cartesian_args(domain, meshing_args, {})
+
+        # Then we create a new mdg with the same fractures and physical dimensions as
+        # the old one, but with a higher resolution:
+        nx_cells_new = np.array(nx_cells) * self.params.get(
+            "interface_refinement_ratio", 2
+        )
+        fracs = [self.fractures[i].pts for i in range(len(self.fractures))]
+        mdg_new = pp.meshing.cart_grid(fracs, nx_cells_new, physdims=phys_dims)
+
+        # Loop through all the interfaces in the new mixed-dimensional grid (donor) such
+        # that we can fill intf_map with a map between old and new interface grids.
+        intf_map: dict[
+            pp.MortarGrid,
+            Union[pp.MortarGrid, dict[mortar_grid.MortarSides, pp.Grid]],
+        ] = {}
+        for intf in mdg_new.interfaces(dim=1):
+            # Then, for each interface, we fetch the secondary grid which belongs to it.
+            _, g_sec = mdg_new.interface_to_subdomain_pair(intf)
+
+            # We then loop through all the interfaces in the original grid (recipient).
+            for intf_coarse in self.mdg.interfaces(dim=1):
+                # Fetch the secondary grid of the interface in the original grid.
+                _, g_sec_coarse = self.mdg.interface_to_subdomain_pair(intf_coarse)
+
+                # Checking the fracture number of the secondary grid in the recipient
+                # mdg. If they are the same, i.e., if the fractures are the same ones,
+                # we update the interface map.
+                if g_sec_coarse.frac_num == g_sec.frac_num:
+                    intf_map.update({intf_coarse: intf})
+
+        # Finally replace the subdomains and interfaces in the original
+        # mixed-dimensional grid.
+        self.mdg.replace_subdomains_and_interfaces(sd_map=grid_map, intf_map=intf_map)
```

Comment:
Aren't lines 192 through 252 just a repetition of what happens during call to pp.mdg_library.square_with_orthogonal_fractures?